### PR TITLE
Let StorageNode subscribe to BucketspacesConfig and propagate to stor…

### PR DESCRIPTION
**Pull request by geirst**
…age components.

@vekterli please review

### CODE REVIEW

File: storage/src/vespa/storage/storageserver/storagenode.cpp [hunk 1]
```diff
@@ -501,6 +536,19 @@ void StorageNode::configure(std::unique_ptr<document::DocumenttypesConfig> confi
     }
 }
 
+void
+StorageNode::configure(std::unique_ptr<BucketspacesConfig> config)
+{
+    {
+        vespalib::LockGuard configLockGuard(_configLock);
+        _newBucketSpacesConfig.reset(config.release());
```
Comment (vekterli) [inline storage/src/vespa/storage/storageserver/storagenode.cpp]: `std::move` into `_newBucketSpacesConfig` instead seems cleaner

**Review by vekterli — APPROVED** [review]
:+1: only nitpicks

**Comment by geirst** [comment]
@vekterli PTAL

**Review by vekterli — APPROVED** [review]
:+1:

**Pull request merged**